### PR TITLE
shellEnv platform/terminal -> platform/shell

### LINKED ISSUE #145405: shellEnv doesn't belong in platform/terminal
Moved in 62dc317056408a1044a3631aec83b930226cb48b

I know there's nothing enforcing this, but the terminal folder should rarely be depended upon outside of debug/tasks. Having `shellEnv.ts` in platform/terminal means that platform/request depends on the terminal which creates a weird relationship:

- platform/request -> platform/terminal
- contrib/terminal -> platform/request

Terminal only exists in platform so that it can be used in the pty host, I'd prefer to keep it similar to a workbench contrib where nothing else in platform depends on it.

If no other folder makes sense I propose creating a `platform/shell` as it's a similar theme to what's going on in base/shell.ts. The original `platform/environment` seemed fine but I don't understand the tangle.

### PR DESCRIPTION
**Pull request by Tyriar**
Fixes #145405

### CODE REVIEW

**Comment by meganrogge** [comment]
Since this is strictly debt, should we wait until next week to merge it?

**Review by meganrogge — APPROVED** [review]
Looks small enough

**Comment by Tyriar** [comment]
Should be safe so thought now it fine

**Pull request merged**
